src/USER-BOCS/fix_bocs.cpp

```diff
@@ -630,154 +635,194 @@ void FixBocs::init()
+    error->message(FLERR, message);
+
+    // Allocate memory for the two dimensional matrix
+    // that holds data from the input file.
+    memory->create(data, NUM_INPUT_DATA_COLUMNS, numEntries, "data");
 
     double stdVolumeInterval = 0.0;
     double currVolumeInterval = 0.0;
     // When comparing doubles/floats, we need an Epsilon.
     // The literature indicates getting this value right in the
     // general case can be pretty complicated. I don't think it
     // needs to be complicated here, though. At least based on the
-    // sample data I've seen where the volume values are fairly
-    // large.
+    // sample data I've seen where the volume values are fairly large.
     const double volumeIntervalTolerance = 0.001;
-    n_entries = 0;
-    while( fgets(line,199,fpi)) {
-      ++n_entries;
-      test_sscanf = sscanf(line," %f , %f ",&f1, &f2);
+    int lineNum = 0;  // this value is only for  message
+    int numBadVolumeIntervals = 0; // count these for message
+    float f1, f2;
+    int test_sscanf;
+    for (int i = 0; i < inputLines.size(); ++i) {
+      lineNum++;  // count each line processed now so lineNum messages can be 1-based
+      test_sscanf = sscanf(inputLines.at(i).c_str()," %f , %f ",&f1, &f2);
       if (test_sscanf == 2)
       {
-        data[0][n_entries-1] = (double) f1;
-        data[1][n_entries-1] = (double) f2;
-        if (n_entries == 2) {
-          stdVolumeInterval = data[0][n_entries-1] - data[0][n_entries-2];
+        //message = fmt::format("INFO: f1 = {}, f2 = {}", f1, f2);
+        //error->message(FLERR,message);
+        data[VOLUME][i] = (double)f1;
+        data[PRESSURE_CORRECTION][i] = (double)f2;
+        if (i == 1)
+        {
+          // second entry is used to compute the validation interval used below
+          stdVolumeInterval = data[VOLUME][i] - data[VOLUME][i-1];
+          //message = fmt::format("INFO: standard volume interval computed: {}", stdVolumeInterval);
+          //error->message(FLERR,message);
         }
-        else if (n_entries > 2) {
-          currVolumeInterval = data[0][n_entries-1] - data[0][n_entries-2];
+        else if (i > 1)
+        {
+          // after second entry, all intervals are validated
+          currVolumeInterval = data[VOLUME][i] - data[VOLUME][i-1];
+          //message = fmt::format("INFO: current volume interval: {}", currVolumeInterval);
+          //error->message(FLERR,message);
           if (fabs(currVolumeInterval - stdVolumeInterval) > volumeIntervalTolerance) {
-            snprintf(badDataMsg,MAX_MESSAGE_LENGTH,
-                     "BAD VOLUME INTERVAL: spline analysis requires uniform"
-                     " volume distribution, found inconsistent volume"
-                     " differential, line %d of file %s\n\tline: %s",
-                     n_entries,filename,line);
-            error->message(FLERR,badDataMsg);
+            message = fmt::format("ERROR: Bad volume interval. Spline analysis requires uniform"
```

Comment:
Since all MPI ranks seem to be reading the file, this message should only be printed on rank 0. Or the reader should be only run on MPI rank 0 and then the data needs to be sent to the other MPI ranks with MPI_Bcast().